community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -116,12 +120,41 @@ public final boolean next( long pageId ) throws IOException
     {
         if ( currentPageId == pageId )
         {
+            verifyContext();
             return true;
         }
         nextPageId = pageId;
         return next();
     }
 
+    void verifyContext()
+    {
+        VersionContext versionContext = versionContextSupplier.getVersionContext();
+        long lastClosedTransactionId = versionContext.lastClosedTransactionId();
+        if ( lastClosedTransactionId == Long.MAX_VALUE )
```

Comment:
Resurfacing comment: The interface should document that this value is special, then.